On AIX, load shared libraries even when created with full aix-soname support:
http://lists.gnu.org/archive/html/libtool/2011-01/msg00023.html
http://lists.gnu.org/archive/html/libtool/2011-02/msg00000.html
Once there is an agreement on that, this patch eventually is for upstream too.

--- subversion/libsvn_ra/ra_loader.c.orig	2009-04-28 14:53:35 +0200
+++ subversion/libsvn_ra/ra_loader.c	2009-04-28 14:55:37 +0200
@@ -164,6 +164,13 @@
 
     /* find/load the specified library */
     SVN_ERR(svn_dso_load(&dso, libname));
+#ifdef _AIX
+    if (! dso) {
+        libname = apr_psprintf(pool, "libsvn_ra_%s-%d.so.0(shr.o)",
+                           ra_name, SVN_VER_MAJOR);
+        SVN_ERR(svn_dso_load(&dso, libname));
+    }
+#endif
     if (! dso)
       return SVN_NO_ERROR;
 
